src/framework/ui/internal/platform/linux/linuxplatformtheme.cpp

```diff
@@ -28,26 +28,53 @@
 using namespace mu::ui;
 using namespace mu::async;
 
+static const QLatin1String service("org.freedesktop.portal.Desktop");
+static const QLatin1String path("/org/freedesktop/portal/desktop");
+static const QLatin1String interface("org.freedesktop.portal.Settings");
+
 LinuxPlatformTheme::LinuxPlatformTheme()
 {
 }
 
 void LinuxPlatformTheme::startListening()
 {
+    QDBusConnection::sessionBus().connect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                          SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isSystemThemeDark = isSystemThemeCurrentlyDark();
+    m_isListening = true;
 }
 
 void LinuxPlatformTheme::stopListening()
 {
+    QDBusConnection::sessionBus().disconnect(service, path, interface, QStringLiteral("SettingChanged"), this,
+                                             SLOT(processSettingChange(QString,QString,QDBusVariant)));
+    m_isListening = false;
 }
 
 bool LinuxPlatformTheme::isFollowSystemThemeAvailable() const
 {
-    return false;
+    const QDBusConnection connection = QDBusConnection::sessionBus();
+
+    static const QString registrarService = QStringLiteral("org.freedesktop.portal.Desktop");
+    const auto iface = connection.interface();
+    if (!iface->isServiceRegistered(registrarService)) {
+        return false;
+    }
+
+    auto message = QDBusMessage::createMethodCall(service, path, interface, QStringLiteral("Read"));
+    message << QStringLiteral("org.freedesktop.appearance") << QStringLiteral("color-scheme");
+
+    QDBusReply<QVariant> reply = QDBusConnection::sessionBus().call(message);
```

Comment:
Maybe it's a good idea to extract these lines into separate method that returns `QDBusReply<QVariant>`, and reuse it in `isSystemThemeCurrentlyDark()`, to prevent the duplication. 

Also, I see multiple references to `QStringLiteral("org.freedesktop.appearance")` and `QStringLiteral("color-scheme")`, maybe we should create `static const QStrings` for those.